Server/BackEndMonitoringServer/src/CContainerOfLogicalDisk.cpp

```diff
@@ -43,39 +43,21 @@ bool CContainerOfLogicalDisk::TryGetAllExistedLogicalDisksAndInfo()
 		// will be changed after implementing an exception handler
 		return false;
 	}
-	const unsigned short c_size_of_buffer_for_api = 1024;
-	//We just skip some chars
-	const unsigned short number_of_chars_need_miss = 1U;
-	char container_all_disks_names[c_size_of_buffer_for_api +
-		c_size_of_buffer_for_api] = {};
-
-	if (PlatformUtils::TryGetLogicalDisksNames(container_all_disks_names,
-		c_size_of_buffer_for_api))
-	{
-		char* variable_for_checking_names = container_all_disks_names;
+	std::vector<std::string> all_names_of_disks;
 
-		while (*variable_for_checking_names)
+	if (PlatformUtils::TryGetLogicalDisksNames(all_names_of_disks))
+	{
+		for (const auto& disk_name : all_names_of_disks)
 		{
-
-			std::string name_of_disk = variable_for_checking_names;
-
-			if (!Utils::TryGetFormattedDiskName(name_of_disk))
-			{
-				return false;
-			}
 			CLogicalDiskInfo* created_disk = new CLogicalDiskInfo();
```

Comment:
C-pointer in 2k21 XD Think about smart-pointers.

Reply:
I will try to change all pointers to smart-pointer, but in another pull-request